clients/src/main/java/org/apache/kafka/common/Cluster.java

```diff
@@ -250,8 +226,9 @@ public Integer partitionCountForTopic(String topic) {
      * @return A list of partitions
      */
     public List<PartitionInfo> availablePartitionsForTopic(String topic) {
-        List<PartitionInfo> parts = this.availablePartitionsByTopic.get(topic);
-        return (parts == null) ? Collections.emptyList() : parts;
+        return partitionsByTopic.get(topic).stream()
```

Comment:
My only concern about this is that it is called on every send() in `DefaultPartitioner`. I think we may have to keep these indexes into partition info.

Reply:
Added back the available partition index.